src/clparser.cc

```diff
@@ -53,15 +53,15 @@ string CLParser::FilterShowIncludes(const string& line,
 
 // static
 bool CLParser::IsSystemInclude(string path) {
-  transform(path.begin(), path.end(), path.begin(), ::tolower);
+  transform(path.begin(), path.end(), path.begin(), [](const char c) { return static_cast<char>(::tolower(c)); });
```

Comment:
MS VC compiler has afaik no switches to select specific c++ versions. Regardless which c++ version is used in ninja, it is not compilable with Visual Studio 2017 and needs to be fixed. 

Any suggestions?